### ISSUE TITLE
DOUBLE EJECTIONS!!!

### ISSUE THREAD

**Issue by iamwhoiamhahaha** [hanabi/hanabi.github.io#761]
First, check my issues I posted before:
- [Double Play Double Bluff (or Double Ejection) & Double Play Ejection clarification](https://github.com/hanabi/hanabi.github.io/issues/90) on Nov 2, 2019
- [Self Color Double Ejection](https://github.com/hanabi/hanabi.github.io/issues/440) on Dec 17, 2020

Secondly, let's see the Discord chat yesterday:

> pianoblook (he/they) — Yesterday at 12:47 PM
lolol, I guess it could. But I've always drawn the line at that level of context. @IAMJEFF has proposed some spicy stuff like that, e.g. Double Ejections

> IAMJEFF — Yesterday at 9:41 PM
(Unnecesary) Poke/Replay/Trash Double Ejections
> - If Bob sees that a Poke/Replay/Trash Double Ignition cannot be true, i.e, Cathy doesn't have a playable or a one-away card on slot 1, then Bob should eject as a Poke/Replay/Trash Ejection.
> - Now, it's Cathy's turn. She can see that Alice could just give a direct Play Clue on the ejected card. The ejection was unnecessary!
> - However, if Cathy has a playable card (or a one-away card) on slot 1, Bob would just blind-play his slot 1 as a Double Iginition, which implies the focus shifts by one slot, i.e., slot 2. Thus, Cathy blind-plays her slot 2 as a Double Ejection to solve the Unnecessary.

> Stephen — Today at 7:07 AM
fwiw I privately proposed this a while back, so it's clearly fairly "natural" if multiple people are coming up w/ the same thing. But my judgment is the same as it was back then. It won't make it into H-group conventions proper.

> timotree — Today at 7:10 AM
I too have independently arrived at this convention

I think it's time to add *Double Ejections* to the doc now, since there are some players individually thought about this convention before and it seems "natural" for all of us.

Here are the proposals with replays:

### (Known) Double Ejection
- When a clue is given retouching a two-or-more-away card(s) and the next player can see that it can't be a *Finesse* or a *Double Bluff*, then the next two players have to respond the clue as a *Double Ejection*.
- Example 1: https://hanab.live/replay/624699#4
  - ***"It actually worked during the game."***
  - jeff gives a 4 clue to kimbi
  - it gets kakashi's g1 and rakxer's black 5 (imagine it is playable)
  - the original purpose was to get the g1 and the g2 from slot 4, but they both responded the "correct" slots as a *Double Ejection* rather than a *Double Positional Clue*
- Example 2: https://hanab.live/replay/354097#13
  - jeff gives a 5 clue to zamiel
  - it gets zamiel's black 2 and avander's rainbow 2
- Example 3: https://hanab.live/replay/633079#7
  - kooolant gives a blue clue to helana
  - it might get kimbi's blue 2 and durandal 's red 1
  - but kimbi would just blind-play slot 1 as a finesse
- Example 4: https://hanab.live/replay/640674#6
  - mice reclues the gray pink 5 with a 5 clue
  - it gets valetta's green 1 and kimbi's red 2 (out-of-position)
- Example 5: https://hanab.live/replay/625933#5
  - floriman gives a 5 clue to kimbi
  - it gets kimbi's green 1 and mice's yellow 1

### (Unnecessary) Poke Double Ejection
- Example: (searching)

### (Unnecessary) Replay Double Ejection
- Example 1: https://hanab.live/replay/595094#14
  - timo gives a blue clue to kimbi
  - it gets piano's red 2 and jeff's red 3
- Example 2: https://hanab.live/replay/593365#20
  - jeff gives a blue clue to lanvin
  - it gets lanvin's yellow 2 and melwen's red 3
  - OR lel's red 3 as *Unnecessary Replay Ejection*
- Example 3: https://hanab.live/replay/658573#4
  - kimbi gives a blue clue to lanvin or 2 to hoodie
  - it gets hoodie's yellow 1 (from slot 2) and frosty's red 1
- Example 4: https://hanab.live/replay/616116#8
  - markus gives a 1 clue to kyle
  - it gets kimbi's light pink 2 and lanvin's prism 1
  - the double ejection is better than the 1 clue to lanvin because the team could give a red clue afterward (on the red 1 and red 4) to protect the red 4 at the same time
- Example 5: https://hanab.live/replay/674556#14
  - orucsoraihc gives a yellow clue to piano
  - it gets piano's red 4 and eliclax's purple 3
- Example 5: https://hanab.live/replay/585126#17
  - timo gives a red clue to kimbi
  - it gets piano's g3 and jeff's b1

### (Unnecessary) Trash Double Ejection
- Example 1: https://hanab.live/replay/672647#29
  - frosty gives a 1 clue to jeff
  - it gets omegaxis' m5 and newduke's b2
- Example 2: https://hanab.live/replay/674556#41
  - eliclax gives a 1 clue to piano
  - it gets orucsoraihc's yellow 4 and piano's green 3
- Example 3: https://hanab.live/replay/52138#58
  - squirtle gives a 4 clue to bulbasaur
  - it gets bulbasaur's red 5 and charmander's rainbow 5

**Comment by iamwhoiamhahaha** [comment]
wait, why is it stale?

**Comment by Zamiell** [comment]
because noone said that they thought it was a good idea in 20 days

**Comment by iamwhoiamhahaha** [comment]
it means noone disagrees with this proposal

**Comment by Zamiell** [comment]
https://en.wikipedia.org/wiki/Null_hypothesis

**Comment by iamwhoiamhahaha** [comment]
piano reacted with an emoji though

**Comment by pianoblook** [comment]
I should have said something other than just a reaction - and I hope others do too!

In general I think this is good, and probably the most intuitive way to play this sort of _Unnecessary Ejection_. As Jeff's awesome examples show, all of this just boils down to Cathy's perspective: "if my finesse was playable, Bob wouldn't have _Ejected_ --> therefore my _Unnecessary_ component must come from deeper in hand".

There is a weird edge-case that might come up, though: what if Cathy's slot 1 is simply the connecting card to what Bob just _Ejected_, and there's just context for Ejecting versus slot 1? The simplest example would be the following:

All 1s played, Black 6 Suit
B: x b2 x x 
C: b3 k3 x x
D: x x x [k5]
> Alice reclues 5 to Donald
> Bob considers this clue; if he truly had b2 on Finesse, Alice would simply clue 3s to Cathy.
> Therefore, Bob _Ejects_
> Cathy can see this was _Unnecessary: no new positive info was gained, and b2 was directly cluable.

...And that's when Cathy needs to decide; or rather, that's where our conventions need to decide: does she play slot 1 and trust there was context, or do we prescribe a _Double Ejection_? My bet is that the Double Ejection will end up being much more useful, but I could see it being confusing.

For the record I'm adding these to the KiPiPi doc to experiment with, specifically because of all the nice examples. We've hesitated in adopting these in the past because tbh we were afraid they'd be too hard to make work, but I do think they're probably objectively better than the alternative.

**Comment by iamwhoiamhahaha** [comment]
> does she play slot 1 and trust there was context

That was what I have proposed in this [issue](https://github.com/hanabi/hanabi.github.io/issues/401) as a *Double Bluff Ejection Finesse*. But I like **double ejection** interpretation more than **ejection finesse** because the latter seems very confusing I guess.

**Comment by pianoblook** [comment]
Ahh yes, thanks for linking that. I remember the Ejection+Finesse being a step too context-dependent to want to implement, personally. Cool then, I also like this revision - the Double Ejection does feel natural. 

In fact, speaking of old proposals, I just remembered that the _Ignition_ portion of this reminds me of #449, though it's not actually the same. I like this more because it still leaves the door open for more _Necessary Ejections_ for tricky cards in Bob's hand, even in cases where Cathy also has a playable on slot 2/3/etc

**Comment by iamwhoiamhahaha** [comment]
got another example for *Trash Double Ejection* in my old replays:
https://hanab.live/replay/52138#58

**Comment by iamwhoiamhahaha** [comment]
idea from kakashi:
alice is allowed to perform a double ejection by touching a brand-new card if bob can see that cathy doesn't have a playable card on slot 1 but slot 2

**Comment by mmelwen** [comment]
I like this idea  of (Unnecessary) Replay Double Ejection
actually happened yesterday 
![image](https://user-images.githubusercontent.com/66402571/153304631-2582f1d3-3ac5-4b3a-900a-099c8ff67858.png)
to get b1 and u1 both from slot 2, I should eject because Lana's slot 1 is not playable and Lana should play slot 2 from the Unnecessary part. I think it makes perfect sense

**Comment by mmelwen** [comment]
r2 was already prompted by the red finesse on r4 

I should have included the replay as well, sorry 
https://hanab.live/shared-replay/720948#10

**Comment by Zamiell** [comment]
im not talking about r2, im talking about w2

**Comment by Zamiell** [comment]
sorry, i meant that markus can be expecting you to play u2 and helana to play u3

**Comment by Zamiell** [comment]
sorry, i mixed up white and null

**Comment by mmelwen** [comment]
The thing is that Lana now knows that her slot 1 is not playable and that Markus could just clue me 1, so i make sense the unessesary to go on the next slot. 
I dont like this with the one away, I agree that we removed from the 4charm as well because the probability of having exactly the connecting card is much lower than any playable card

**Comment by iamwhoiamhahaha** [comment]
Now *Self Color Ejection* has been deleted, it leaves the spot for the first form of the *Double Ejections* I proposed 😋

**Comment by iamwhoiamhahaha** [comment]
https://hanab.live/replay/741666#21

**Comment by iamwhoiamhahaha** [comment]
time to make this proposal approve please~
double ejections are powerful
https://hanab.live/replay/756585#53

**Comment by iamwhoiamhahaha** [comment]
https://discord.com/channels/140016142600241152/456214043351580674/1001593876831805510

crossproduct:

> Reading up on some Advanced Strategies 😄
> https://hanabi.github.io/docs/extras/ejections#the-replay-ejection
> 
> Curious about something... If Alice gives something that looks like a Replay Double Ignition, but it can't be (because Cathy's Finesse is not playable), Bob will eject. The doc calls this a Replay Ejection. If the Replay Ejection is unnecessary, will Cathy also Eject? (Unnecessary moves should get an extra blind play, and Cathy can't play Finesse. Assuming Donald, etc. also cannot play Finesse)

Sturm:

> It is not an official convention but depending on the who your teammates are they will know what is up and you will get a double ejection.

**Comment by iamwhoiamhahaha** [comment]
https://discord.com/channels/140016142600241152/465962599851491340/937130002493870151

ace201 — 01/30/2022

> @IAMJEFF https://hanab.live/replay/711015#34
> Double ejection. Probably the first one I pulled off successfully.

**Comment by argothiel** [comment]
I agree. Unnecessary ejection -> ejection looks like a very intuitive idea.

**Comment by iamwhoiamhahaha** [comment]
![doodles](https://user-images.githubusercontent.com/50160641/182006146-439a1d25-ce56-4557-8805-c8dc23870214.png)

**Comment by Dr-Kakashi** [comment]
Hi guys, so I have talked with Jeff over DM's. Instead of me just diving into the detailed analysis, I'm just going to voice my thoughts. 

From what I'm seeing by the examples, Alice is initiating a double bluff. 
Bob can see it is a double bluff, but Bob can also see that Cathy does not have a playable on slot 1. 

In my mind, Bob should just eject and the situation is resolved immediately as Cathy now knows her slot 1 is not playable. I haven't played in a while, isn't this already a convention? 

If not, then the group needs to decide on what to do. 

1. If it is "good enough" for bob to Eject. 
2. If not, then Cathy needs to additionally eject, as she knows her slot 1 is not playable.

Jeff tried to point out to me that Cathy can see the situation is "unnecessary" due to Bob's slot 2 being directly clue-able. The problem I have with this is that it does not meet the unnecessary definition as the examples given has the clue receiver receiving more information to their hand.  Thus, Cathy is now in a position of deciding whether or not it's "bad" enough which is a subjective opinion. 

We have hard definitions in this group to ensure a move is unnecessary or sub-optimal. 

In conclusion, I'm in favor of Bob blind playing by ejecting to immediately resolve the situation. I like clarity and following the minimum blind play principle. There's a potential risk of confusion by having Cathy additionally eject.

**Comment by Dr-Kakashi** [comment]
After talking with Jeff some more....I really feel he is putting Cathy in a subjective situation. 

I have 2 thoughts:
1. How often would a situation like this occur
2. Does these situations outweigh the situations where Alice is trying to give extra information (positive and negative) to the team with the intention of only having Bob eject?

**Comment by timotree3** [comment]
It's only going to be positive information on cards that were already known playable/trash, so it's not very valuable. Negative information is always irrelevant for this kind of thing; trash bluffs gives negative info, but we still treat them as unnecessary. I think it's going to be pretty clear when a Replay/Poke Ejection is unnecessary

**Comment by timotree3** [comment]
To be clear, is the version of this where the reclue is on a 2-away card being proposed still? I guess I was assuming that this was just about Replay/Poke double ejection

**Comment by iamwhoiamhahaha** [comment]
> To be clear, is the version of this where the reclue is on a 2-away card being proposed still?

I think so. But I agree that *(Known) Double Ejection* is not very clear in some situation. We could just make *Replay/Poke Double Ejections* get approved first

**Comment by pianoblook** [comment]
> We could just make _Replay/Poke Double Ejections_ get approved first

+1 to this. It'll be a pretty rare convention (and kinda strange/new to most), so I think we should keep it as clear as possible (at least for now).

**Comment by iamwhoiamhahaha** [comment]
**(Unknown) Trash Double Ejection** example:
https://discord.com/channels/140016142600241152/1007782978610528277/1008279751162593290

**Comment by Zamiell** [comment]
> +1 to this. It'll be a pretty rare convention (and kinda strange/new to most), so I think we should keep it as clear as possible (at least for now).

ok, so how does everyone else feel about this?

Jeff, can you submit a PR?

**Comment by iamwhoiamhahaha** [comment]
done https://github.com/hanabi/hanabi.github.io/pull/976
